virttest/test_setup.py

```diff
@@ -1301,6 +1300,30 @@ def get_pf_ids(self):
             pf_ids.append(pf_id)
         return pf_ids
 
+    def assign_static_ip(self):
+        """
+        Set the static IP for the PF devices for mlx5_core driver
+        """
+        # This function assigns static IP for the PF devices
+        set_ip = 0
+        pf_devices = self.get_pf_ids()
+        if (not self.start_addr_PF) or (not self.net_mask):
+            raise exceptions.TestSetupFail(
+                "No IP / netmask found, please populate starting IP address for PF devices in configuration file")
+        ip_addr = netaddr.IPAddress(self.start_addr_PF)
+        set_ip = 1
+        for PF in pf_devices:
+            if set_ip:
```

Comment:
In your new re-factor I don't think `set_ip` is required.

Reply:
@balamuruhans Addressed the changes. Please review.